lib/secure_headers.rb

```diff
@@ -97,15 +97,10 @@ def ensure_security_headers options = {}
         warn "[DEPRECATION] secure_headers ruby 1.8.7 support will dropped in the next release"
       end
       self.secure_headers_options = options
-      before_filter :prep_script_hash
-      before_filter :set_hsts_header
-      before_filter :set_hpkp_header
-      before_filter :set_x_frame_options_header
-      before_filter :set_csp_header
-      before_filter :set_x_xss_protection_header
-      before_filter :set_x_content_type_options_header
-      before_filter :set_x_download_options_header
-      before_filter :set_x_permitted_cross_domain_policies_header
+      hook = respond_to?(:before_action) ? :before_action : :before_filter
+      [:prep_script_hash, :set_security_headers].each do |method|
+        send(hook, method)
+      end
```

Comment:
The reason for setting the different `before_filter`s is that it makes it possible to use `skip_before_filter` for each individual header. Calling `set_security_headers` would remove this granularity.